src/App.tsx

```diff
@@ -8,7 +8,68 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+
+enum Filter {
+  ALL = 'all',
+  ACTIVE = 'active',
+  COMPLETED = 'completed',
+}
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setLoading] = useState(true);
+  const [isModal, setModal] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filter, setFilter] = useState<Filter | string>(Filter.ALL);
+  const [filteredTodos, setFilteredTodos] = useState(todos);
+  const [searchQuery, setSearchQuery] = useState('');
+
+  useEffect(() => {
+    getTodos()
+      .then(res => {
+        setTodos(res);
+        setLoading(false);
+      });
+  }, []);
+
+  useEffect(() => {
+    switch (filter) {
+      case Filter.ACTIVE:
+        setFilteredTodos(todos.filter(todo => !todo.completed));
+
+        break;
+      case Filter.COMPLETED:
+        setFilteredTodos(todos.filter(todo => todo.completed));
+
+        break;
+      default:
+        setFilteredTodos(todos);
```

Comment:
it's better to add `case Filter.ALL`, it will improve readability and help us when we want to scale our app